### PR TITLE
feat: initial identity setup

### PR DESCRIPTION
**Pull request by clowenhg**
## Summary by CodeRabbit

- **New Features**
    - Enhanced security and configuration management by parsing OAuth configuration from environment variables.
    - Implemented OAuth2 initialization and identity management features with JWT token generation and verification.
    - Enabled users to authenticate via Discord and receive a JWT token for identity verification.

- **Bug Fixes**
    - Improved handling of missing environment variables during configuration setup to prevent runtime errors.

### CODE REVIEW

File: registry/cmd/config.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			AuthorizationURL: authURL,
+			TokenURL:         tokenURL,
+			RedirectURL:      redirectURL,
+			RevokeURL:        revokeURL,
+			ClientID:         clientID,
+			ClientSecret:     clientSecret,
+		},
+	}
+
+	return cfg, nil
```
[the remark this answers is not in the document]
Comment (clowenhg) [reply registry/cmd/config.go:95]: Ignore this issue.
`errors.Join` is part of builtin package.

[the remark this answers is not in the document]
Comment (harjotgill) [reply registry/cmd/config.go:95]: @coderabbitai please remember this fact for the future. `errors.Join` is now part of the standard library.

File: registry/internal/identity/oauth/oauth.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	http.HandleFunc("/oauth/callback", s.oAuthCallback)
+	http.HandleFunc("/login", func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, url, http.StatusFound)
+	})
+
+	srv := &http.Server{
+		Addr: ":8080",
+	}
+	log.Fatal(srv.ListenAndServe())
```
[the remark this answers is not in the document]
Comment (clowenhg) [reply registry/internal/identity/oauth/oauth.go:31]: @coderabbitai help

[the remark this answers is not in the document]
Comment (clowenhg) [reply registry/internal/identity/oauth/oauth.go:31]: @coderabbitai can you suggest a fix.

**Pull request merged**